src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -13,69 +15,80 @@
 import java.util.List;
 
 @RestController
-@RequestMapping("/products")
+@RequestMapping("/product")
 @Slf4j
 @RequiredArgsConstructor
 public class ProductController {
     private final ProductService productService;
+    private final ProductMapper productMapper;
 
     @PostMapping("/")
     @PreAuthorize("hasRole('SELLER')")
-    public ResponseEntity<ProductEntity> createProduct(
-            @ModelAttribute("name") String name,
-            @ModelAttribute("thumbnailImage") MultipartFile thumbnailImage,
-            @ModelAttribute("image") MultipartFile image,
-            @ModelAttribute("available") Boolean available,
-            @ModelAttribute("description") String description,
-            @ModelAttribute("categoryId") Long categoryId,
-            @ModelAttribute("sellerId") Long sellerId) throws IOException {
-
-        ProductEntity newProduct = productService.createProduct(name, thumbnailImage, image, available, description, sellerId, categoryId);
-
-        return new ResponseEntity<>(newProduct, HttpStatus.CREATED);
+    @ResponseStatus(HttpStatus.CREATED)
+    public ProductResponse createProduct(
+            @Valid
+            @ModelAttribute ProductCreateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
+
+        ProductEntity newProduct = productService.createProduct(request, thumbnailImage, image);
+
+        return productMapper.toProductResponse(newProduct);
     }
 
     @PatchMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<ProductEntity> patchProduct(@PathVariable Long productId,
-                                                      @ModelAttribute ProductUpdateRequest request,
-                                                      @RequestParam("thumbnailImage") MultipartFile thumbnailImage,
-                                                      @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse patchProduct(
+            @Valid
```

Comment:
does `@Valid` apply to `@PathVariable Long productId` only  or all args?
will it apply to `@ModelAttribute ProductUpdateRequest request`?

Reply:
Only ProductUpdateRequest because it has annotations for validation.

Reply:
ok, please check how this works in Postman:
![image](https://github.com/user-attachments/assets/0988fdd1-d60d-4316-9ea9-7a598abf16ff)

if it works as expected, then you should get en exception on validation: 
`@NotBlank(message = "Product name cannot be blank")`

if it doesn't work properly, that means you are trying to validate `@PathVariable Long productId` instead.

Reply:
Moved annotation @Valid from @ModelAttribute.